src/components/GraphLayout/GraphLayout.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+import { useState } from "react";
+import Button from "../Button/Button";
+import styles from "./GraphLayout.module.css";
+import RangeInput from "../RangeInput/RangeInput";
+import DiceSvg from "../../../public/svg/DiceSvg";
+import SortSvg from "../../../public/svg/SortSvg";
+import Stalin from "../../../public/Stalin.jpg";
+import Image from "next/image";
+function GraphLayout({
+  sorrtingArray,
+  title,
+  onRandomize,
+  onSort,
+  isAbleToSort,
+  chosenColumn,
+  comparingColumn,
+  sortingSpeedFunction,
+  stalinKills,
+}: {
+  sorrtingArray: number[];
+  title: string;
+  onRandomize: () => void;
+  onSort: () => void;
+  isAbleToSort: boolean;
+  chosenColumn?: number | number[];
+  comparingColumn?: number;
+  sortingSpeedFunction: (sortingSpeed: number) => void;
+  stalinKills?: number[];
+}) {
+  const handleRandomize = async () => {
+    onRandomize();
+  };
+
+  const handleSort = async () => {
+    onSort();
+  };
+
+  const [sliderValue, setSliderValue] = useState<number>(50);
+
+  const pullSliderValueFromChild = (sliderValue: number) => {
+    setSliderValue(sliderValue);
+  };
+  sortingSpeedFunction(sliderValue);
+  const [isStalinActivated, setIsStalinActivated] = useState<boolean>(false);
+  return (
+    <div className={styles.GraphLayout}>
+      <div className={styles.title}>
+        <div
+          onClick={() => {
+            title === "Stalin Sort"
+              ? setIsStalinActivated((prevComrad) => !prevComrad)
+              : null;
+          }}
+          className={styles.titleName}
+        >
+          {title}
+        </div>
+      </div>
+      <div
+        className={styles.graphSection}
+        style={{
+          backgroundImage:
```

Comment:
And also check if next does not implement any optimizations for images as background